2.0/m_joinpartspam.cpp

```diff
@@ -188,6 +199,21 @@ class ModuleJoinPartSpam : public Module
 		ServerInstance->Modules->SetPriority(this, I_OnUserJoin, PRIORITY_LAST);
 	}
 
+	void OnRehash(User*)
+	{
+		/* Check for a channel redirect value and send an Announcement SNOTICE
+                 * if it's invalid or the ban redirect module is not loaded */
+		ConfigTag* tag = ServerInstance->Config->ConfValue("joinpartspam");
+		const std::string redirchan = tag->getString("redirchan");
+		if (!redirchan.empty())
+		{
+			if (!ServerInstance->IsChannel(redirchan.c_str(), ServerInstance->Config->Limits.ChanMax))
```

Comment:
Rather than sending a snotice here and in the line below you would probably be better off throwing ModuleException to prevent the module from loading.